src/main/java/io/electrum/airtime/api/PurchaseResource.java

```diff
@@ -156,6 +164,30 @@ public final void reversePurchase(
             .reversePurchase(body, securityContext, request, httpHeaders, asyncResponse, uriInfo, httpServletRequest);
    }
 
+   @POST
+   @Path(RetryPurchase.RELATIVE_PATH)
+   @Consumes({ "application/json" })
+   @Produces({ "application/json" })
+   @ApiOperation(nickname = RetryPurchase.RETRY_PURCHASE, value = "Retry a previously submitted purchase request.",
+           notes = "If no response was received to a purchase request due to a timeout or temporary communications failure, PoS may retry the same purchase request by calling this resource. The original purchase request will be resubmitted to the provider. If the provider had received the original request, it will respond by returning any bundles or airtime adjustments that were already issued. If not, then either new bundles or airtime adjustments may be issued as per a normal purchase or the retry will be declined.")
```

Comment:
I think a natural question is "when do I do a reversal and when do I attempt a retry?" to which the answer is probably "This operation is intended primarily for purchases which do not support refunds. For example, direct top-up transactions (a.k.a. PIN-less transactions) often do not support reversals. In such instances, a retry may be used instead of a reversal to guarantee a definite and final outcome."

Reply:
I've appended this to the notes as follows: Let me know if we need to change it.
```
 "If no response was received to a purchase request due to a timeout or temporary communications "
         + "failure, PoS may retry the same purchase request by calling this resource. The original purchase "
         + "request will be resubmitted to the provider. If the provider had received the original request, "
         + "it will respond by returning any bundles or airtime adjustments that were already issued. If not, "
         + "then either new bundles or airtime adjustments may be issued as per a normal purchase or the retry "
         + "will be declined. This operation is intended primarily for purchases which do not support refunds. "
         + "For example, direct top-up transactions (a.k.a. PIN-less transactions) often do not support "
         + "reversals. In such instances, a retry may be used instead of a reversal to guarantee a definite "
         + "and final outcome."
```

Reply:
> `This operation is intended primarily for purchases which do not support refunds.`

There's a big difference between a refund and a reversal. I think you meant:
```suggestion
This operation is intended primarily for purchases which do not support reversals.
```
If you _did_ mean "refunds" then let's chat.

Reply:
✅ changed "refunds" to "reversals".

Removed the sentence about the service provider's actions:
```
notes = "If no response was received to a purchase request due to a timeout or temporary communications "
         + "failure, PoS may retry the same purchase request by calling this resource. The original purchase "
         + "request will be resubmitted to the provider. "
         + "This operation is intended primarily for purchases which do not support reversals. "
         + "For example, direct top-up transactions (a.k.a. PIN-less transactions) often do not support "
         + "reversals. In such instances, a retry may be used instead of a reversal to guarantee a definite "
         + "and final outcome."
```
I agree that the description would not have made sense/been accurate, especially after the handling of retries are changed to be the same as in the PPU server.